### PR TITLE
Cache blockly flyouts for faster subsequent loads (~3X speedup)

### PR DESCRIPTION
**Pull request by darzu**
REQUIRES: https://github.com/Microsoft/pxt-blockly/pull/208

Previously, we would recompute and relayout all the blocks each time we opened the "flyout" when clicking on the toolbox.

With this change, we maintain separate flyouts for each toolbox category and toggle their visiblity state when switching between categories. This doesn't speed up the initial load, but subsequent loads are ~3X faster.

I'm also investigating a way to get another 3X (for a total of 9X) speed up by playing nice with React so it doesn't force a style recompute when toggling visibility.

Note this affects all blockly editors.

## Before
[macbookpro, x6 slowdown] 1st load: ~1.3 seconds
[macbookpro, x6 slowdown] nth load: ~600 ms
[chrome book LC330] 1st load: ~1.8 seconds
[chrome book LC330] nth load: ~680ms
flyouts_perf_before

## After
[macbookpro, x6 slowdown] 1st load: ~1.3 seconds (same)
[macbookpro, x6 slowdown] nth load: ~180 ms
flyouts_perf_after

perf numbers are from Chrome profiler using: 
- 6x CPU throttling on MacBook Pro 2018 i9
- Lenovo C330 Chrome book w/ MT8173C

### CODE REVIEW

**Comment by pelikhan** [comment]
Do you clear the cache when the header gets reloaded?

**Comment by darzu** [comment]
> Do you clear the cache when the header gets reloaded?

What do you mean by "header"?

**Comment by pelikhan** [comment]
When a script gets reloaded. Looks like you do.

See “loadHeaderAsync”

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: localtypings/blockly.d.ts [hunk 1]
```diff
@@ -1369,25 +1369,6 @@ declare module Blockly {
              */
             setCommentText(text: string): void;
     
-            /**
-             * Set this Block's breakpoint as enabled or disabled.
-             * @param {boolean} enable A boolean definining if the breakpoint should be enabled or disabled.
-             */
-            enableBreakpoint(enable: boolean): void;
```
Comment (pelikhan) [inline localtypings/blockly.d.ts]: did those api dissapear? make sure to work of the "develop" branch in blockly.

File: localtypings/blockly.d.ts [hunk 2]
```diff
@@ -13453,6 +13436,14 @@ declare module Blockly.Functions {
...
      */
     function isFunctionArgumentReporter(block: Blockly.BlockSvg): boolean;
+
+    /**
+     * Create a flyout, creates the DOM elements for the flyout, and initializes the flyout.
+     * @param {!Blockly.Workspace} workspace The target and parent workspace for this flyout. The workspace's options will
+     *     be used to create the flyout's inner workspace.
+     * @return { flyout: !Blockly.Flyout, flyoutSvg: SVGSVGElement } True if the block is a function argument reporter.
+     */
+    function createFlyout(workspace: Blockly.Workspace): void;
```
Comment (pelikhan) [inline localtypings/blockly.d.ts]: return type incorrect

**Comment by darzu** [comment]
Flyout perf improvements testable here: https://arcade.staging.pxt.io/app/ccc806ea325942a07d3e7b76c24dfbbc41876c9a-e6714039f6

**Comment by darzu** [comment]
> Ready to go?

Not yet. The staging deployment I made of these changes is failing with:
> {"message":"Internal server error (catch)"}
and I want to make sure it's not related to my changes.

I'll merge when rdy

**Comment by darzu** [comment]
Testable here: https://arcade.makecode.com/app/c641ae5a704f37f9bfe91a1f4f388e4fb56b94e2-df2a70577f#editor

**Pull request merged**